scw_serverless/gateway/gateway_manager.py

```diff
@@ -1,8 +1,17 @@
+from typing import Protocol
+
 import scaleway.function.v1beta1 as sdk
 from scaleway import Client
 
 from scw_serverless.app import Serverless
-from scw_serverless.gateway.gateway_api_client import GatewayAPIClient
+from scw_serverless.config.route import GatewayRoute
+
+
+class Gateway(Protocol):
```

Comment:
Might be a bit anti YAGMI to use a protocol when we have a single implementation, it's only useful for the unit tess.